contrib/pkg/input/input_reconciler.go

```diff
@@ -130,7 +130,8 @@ func (r *inputReconciler) processNextWorkItem() bool {
 		requeue bool
 		err     error
 	)
-	if isRemoteCluster {
+	// TODO (ilackarms): this is a workaround for an issue where Relay sets the clusterName field of an object, but the underlying reconciler is a single-cluster reconciler.
```

Comment:
Is it no longer true that `clusterName` is only populated for multicluster reconcilers?

Could we modify relay so that this convention continues to hold?

Reply:
i want to have relay set the cluster name on objects because we are building snapshots of objects that come from remote clusters (e.g. cert issuer) where we rely on cluster name to process the object in the translator

Reply:
the issue is that we run a single-cluster reconciler to reconcile on local events, but also want to trigger the same reconcile function when a remote object changes (instead of just running a multi cluster reconciler)